src/main/java/com/checkmarx/flow/custom/GitLabIssueTracker.java

```diff
@@ -71,19 +70,18 @@ public void init(ScanRequest request, ScanResults results) throws MachinaExcepti
         }
     }
 
-    private Integer getProjectDetails(String repoName) {
+    private Integer getProjectId(String targetNamespace, String targetRepoName) {
         try {
-            String url = properties.getApiUrl().concat(PROJECT);
-            url = url.replace("{repo}", repoName);
-            URI uri = new URI(url);
-            HttpEntity httpEntity = new HttpEntity<>(createAuthHeaders());
-            ResponseEntity<String> response = restTemplate.exchange(uri, HttpMethod.GET, httpEntity, String.class);
-            JSONArray arr = new JSONArray(response.getBody());
-            for(Object obj: arr){
-                JSONObject realObj = (JSONObject) obj;
-                String name = realObj.getString("name");
-                if(name.equals(repoName)) {
-                    return realObj.getInt("id");
+            JSONArray candidateProjects = getProjectSearchResults(targetRepoName);
+            log.debug("Projects found: {}. Looking for exact match.", candidateProjects.length());
+
+            // The search is fuzzy, so we need to additionally filter search results here for strict match.
+            for (Object project : candidateProjects) {
+                JSONObject projectJson = (JSONObject) project;
+                if (isTargetProject(projectJson, targetNamespace, targetRepoName)) {
+                    int result = projectJson.getInt("id");
+                    log.debug("Using GitLab project ID: {}", result);
+                    return result;
```

Comment:
init the result to 0, use break to exit this loop and return the result outside the loop (remove a return statement).

Reply:
Done.